packages/ra-core/src/form/useChoices.ts

```diff
@@ -64,6 +65,12 @@ const useChoices = ({
                     ? optionText(choice)
                     : get(choice, optionText);
 
+            if (isValidElement<{ record: any }>(choiceName)) {
```

Comment:
Not sure why you added that. What's the use case?

Reply:
Can't find the usecase anymore. I'll remove it